src/lib/Lib/ExecutionContext.hpp

```diff
@@ -28,23 +34,88 @@ namespace mrdox {
     instances.
 */
 class ExecutionContext
-    : public tooling::ExecutionContext
 {
-    llvm::sys::Mutex mutex_;
-    Diagnostics diags_;
+protected:
+    const ConfigImpl& config_;
 
 public:
-    explicit
     ExecutionContext(
-        tooling::ToolResults* Results)
-            : tooling::ExecutionContext(Results)
+        const ConfigImpl& config)
+        : config_(config)
     {
     }
 
-    void report(Diagnostics&& diags);
-    void reportEnd(report::Level level);
+    virtual
+    void
+    report(
+        InfoSet&& info,
+        Diagnostics&& diags) = 0;
+
+    virtual
+    void
+    reportEnd(report::Level level) = 0;
```

Comment:
Why is this virtual now?

Reply:
This is only because on this branch there are two different `ExecutionContext` types: one that serializes to bitcode, and one which does not. Thus, I made `reportEnd` virtual just incase their behavior would differ (it does not). This is a intermediate solution which I will be getting rid of eventually.